crates/oxc_transformer/src/typescript/mod.rs

```diff
@@ -109,24 +102,84 @@ impl<'a> TypeScript<'a> {
     /// * Adds `export {}` if all import / export statements are removed, this is used to tell
     /// downstream tools that this file is in ESM.
     pub fn transform_program(&mut self, program: &mut Program<'a>) {
-        let mut needs_explicit_esm = false;
+        let mut export_type_names = FxHashSet::default();
+        let mut export_names = FxHashSet::default();
 
-        for stmt in program.body.iter_mut() {
+        // Collect export names
+        program.body.iter().for_each(|stmt| {
```

Comment:
We already have them inside `ModuleRecord`, I guess these are transformed by previous transformers?

Reply:
Yes, But I worry if other plugins change the imports/exports, the ModuleRecord is not updated. How do you think I could fix this? Like `nodes` / `scopes` / `symbols` do? So it seems that whenever ast changes, we need to update everything in `Semantic`

Reply:
Let's get the tests passing first and refactor later